src/GameServer/Systems/Inventory/InventoryManager.cs

```diff
@@ -55,7 +56,143 @@ private int GetVirtualItemValue(int itemId)
                     return 0;
             }
         }
+        public async Task upgradeWeaponAsync(ulong guid, List<ulong> foodWeaponGuidList, List<ItemParam> itemParamList)
```

Comment:
This function, as well as PromoteWeaponAsync should be delegated to the WeaponTab. Additionally, The function signature should be made more generic:
```suggestion
        public async Task UpgradeWeaponAsync(ulong guid, IEnumerable<ulong> foodWeaponGuidList, IEnumerable<ItemParam> itemParamList)
```